sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -646,9 +646,9 @@ class Analyzer(
         // Since the exprId of extraAggExprs will be changed in the constructed aggregate, and the
         // aggregateExpressions keeps the input order. So here we build an exprMap to resolve the
         // condition again.
-        val exprMap = extraAggExprs.zip(
-          newChild.asInstanceOf[Aggregate].aggregateExpressions.takeRight(
-            extraAggExprs.length)).toMap
+        // We also add the original aggregateExpressions in Map to avoid extraAggExprs is empty.
```

Comment:
what's wrong if `extraAggExprs` is empty?

Reply:
If we resolved the having condition but extraAggExprs is empty, that means the resolution is in original aggregateExpressions.

We need use original aggregateExpressions mapping having condition NamedExpression to the new since grouping sets constructAggregate will build the new alias for it.

Reply:
I don't get it. The comment says
```
        // Since the exprId of extraAggExprs will be changed in the constructed aggregate, and the
        // aggregateExpressions keeps the input order. So here we build an exprMap to resolve the
        // condition again.
```
I don't see why we need to include the original aggregateExpressions in `exprMap`.

Reply:
The comment is right if we not return the resolved having condition when `exprMap` is empty, this pr break the behavior that return it.

If we not add the original aggregateExpressions, we can't mapping the resolved having condition since `exprMap` is empty.

Reply:
can you make the comment clearer?